### PR TITLE
added category model

### PR DESCRIPTION
**Pull request by vadymhrnk**

### CODE REVIEW

File: src/main/resources/db/changelog/changes/06-create-categories-table.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        - createTable:
+            tableName: categories
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (AgroFix) [inline src/main/resources/db/changelog/changes/06-create-categories-table.yaml]: It is not necessary to add nullable: false if you have a primaryKey

File: src/main/resources/db/changelog/changes/07-create-books-categories-table.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      author: vadymhrnk
+      changes:
+        - createTable:
+            tableName: books_categories
+            columns:
+              - column:
+                  name: book_id
+                  type: bigint
+                  constraints:
+                    nullable: false
```
Comment (AgroFix) [inline src/main/resources/db/changelog/changes/07-create-books-categories-table.yaml]: It is not necessary to add nullable: false if you have a primaryKey

**Review by AgroFix — APPROVED** [review]
Good job!!!💪

File: src/main/java/com/example/bookstore/model/Book.java [hunk 3]
```diff
@@ -32,4 +39,11 @@ public class Book {
     private String coverImage;
     @Column(nullable = false)
     private boolean isDeleted = false;
+    @ManyToMany
+    @JoinTable(
+            name = "books_categories",
+            joinColumns = {@JoinColumn(name = "book_id")},
+            inverseJoinColumns = {@JoinColumn(name = "category_id")}
+    )
+    private Set<Category> categories = new HashSet<>();
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/model/Book.java:52]: exclude from `equals/hashCode` and `toString`.

File: src/main/java/com/example/bookstore/model/Category.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Data
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/model/Category.java:15]:
```suggestion
@NoArgsConstructor
@Data
```

File: src/main/java/com/example/bookstore/model/Category.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@SQLRestriction("is_deleted=false")
+@Table(name = "categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    private String description;
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/model/Category.java:27]:
```suggestion
    @Column
    private String description;
```

File: src/main/java/com/example/bookstore/repository/BookRepository.java [hunk 6]
```diff
@@ -1,7 +1,20 @@
...
 
 import com.example.bookstore.model.Book;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/repository/BookRepository.java:13]:
```suggestion
@Repository
public interface BookRepository extends JpaRepository<Book, Long> {
```

File: src/main/java/com/example/bookstore/repository/CategoryRepository.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package com.example.bookstore.repository;
+
+import com.example.bookstore.model.Category;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface CategoryRepository extends JpaRepository<Category, Long> {
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/repository/CategoryRepository.java:8]:
```suggestion
@Repository
public interface CategoryRepository extends JpaRepository<Category, Long> {
```

File: src/main/resources/db/changelog/changes/06-create-categories-table.yaml (hunk 1 above)
Comment (d1sam) [reply src/main/resources/db/changelog/changes/06-create-categories-table.yaml]: agree

File: src/main/resources/db/changelog/changes/07-create-books-categories-table.yaml (hunk 2 above)
Comment (d1sam) [reply src/main/resources/db/changelog/changes/07-create-books-categories-table.yaml]: agree

**Review by d1sam — CHANGES_REQUESTED** [review]
good job, but some changes should be done!

File: src/main/java/com/example/bookstore/repository/BookRepository.java (hunk 6 above)
Comment (vadymhrnk) [reply src/main/java/com/example/bookstore/repository/BookRepository.java:13]: [This](https://stackoverflow.com/questions/65576247/why-put-repository-on-top-of-spring-data-jpa-interfaces) is what I found about annotation, so I guess I need to put it just for clarification?

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**